### PR TITLE
Fix lint and build

### PR DESCRIPTION
**Pull request by sboschman**
### Description of your changes

Bump golangci-lint to be compatible with go 1.18

Removes the native terraform provider binary install, upstream terraform-provider-rancher2 does not seem to be compiled that way. It causes the same build error as https://github.com/crossplane-contrib/provider-jet-template/issues/25

I have:

- [X] Read and followed Crossplane's [contribution process].
- [ ] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested
run `make build`

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

**Comment by sboschman** [comment]
once this is merged I will rebase #9 and #8 @ytsarev

File: Makefile [hunk 1]
```diff
@@ -11,8 +11,13 @@ export TERRAFORM_PROVIDER_VERSION := 1.24.0
 export TERRAFORM_PROVIDER_DOWNLOAD_NAME := terraform-provider-rancher2
 export TERRAFORM_PROVIDER_DOWNLOAD_URL_PREFIX := https://github.com/rancher/terraform-provider-rancher2/releases/download/v$(TERRAFORM_PROVIDER_VERSION)
 
+# Not supported by this terraform provider atm
+export TERRAFORM_NATIVE_PROVIDER_BINARY :=
```
Comment (ytsarev) [inline Makefile]: why is it not supported? Can't we guess/predict the provider name?

Comment (sboschman) [reply Makefile]: The original intent of the native binary change to the Dockerfile by the author is not clear to me. Unless it was a fix for terraform providers bundling multiple binaries in the same zip. But I have not found any evidence of this in the `null` and `azurerm` providers.

As far as I understand it, it boils down to the terraform plugin sdk version used to build a provider and the v4 -> v5 (grpc) protocol changes made from sdk v1 to sdk v2.

terraform-provider-azurerm:
- go.mod: `github.com/hashicorp/terraform-plugin-sdk/v2 v2.18.0`
- published zip with provider binary extracts a `..._v5` binary, a tf plugin/provider using protocol v5 (grpc)

terraform-provider-null (from template):
- go.mod: `github.com/hashicorp/terraform-plugin-sdk/v2 v2.18.0`
- published zip extracts a `...v5` binary

terraform-provider-rancher2:
- go.mod: `github.com/hashicorp/terraform-plugin-sdk v1.14.0`
- published zip extract a binary without any protocol indication (not _v4 nor _v5)

Hence my remark 'native' (grpc or better said protocol v5) is not supported by terraform-provider-rancher2 as it is build against SDK v1 not SDK v2.

But like I said, I am not sure why all this 'native binary' plumbing was introduced in the first place. If you d/l the zip file, it extracts a single binary, the protocol version (v4 or v5 or v6) is not really relevant, just place that single extracted binary in the correct tf plugin dir. Only if tf core drops support for protocol v4 we are stuck with an older tf core version, unless terraform-provider-rancher2 upgrades to SDK v2.

Comment (ytsarev) [reply Makefile]: ok cool, so maybe just remove this variable as it is not used at all?

**Review by ytsarev — APPROVED** [review]
lgtm

**Pull request merged**